### PR TITLE
Prod cluster sizing

### PR DESCRIPTION
**Pull request by mendonk**
* Production cluster sizing methodology page (main page)
* VM-based prod cluster page. 
* Pulsar sandbox page with min requirements.

Current build is here (need VPN): https://coppi.aws.dsinternal.org/streaming-docs/luna-streaming-production-clusters

### CODE REVIEW

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+While Kubernetes is the more popular option, it is easier to express disk calculations, throughput, and secure communications in terms of a VM.
+We will call out Kubernetes specifics as they arise in the guide.
+
+== Pulsar cluster components
+
+Pulsar clusters come in many shapes and sizes. There are minimum components for base functionality, and there are some recommended add-on components that make management and observability easier. For this guide we will focus on the required components and what it takes to make them resilient to outages and highly available in a three-zone cloud.
+
+=== Required components
+
+* Zookeeper - This is Pulsar’s meta data store. It stores data about a cluster’s configuration, helps the proxy direct messages to a broker, and holds Bookie configurations. To star, use 1 instance of Zookeeper in your cluster and let the number of lookups delegate scaling from there.
```
Comment (rodageve) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc]: Typo - 'To Star, use' should be 'To Start'

Comment (rodageve) [reply modules/install-upgrade/pages/production-cluster-sizing.adoc]: You need at least 3 instances of Zookeeper for quorum. Quorum size of (n/2 + 1).  This configuration avoids a 'split-brain' scenario.  I'm not 100% sure if you can deploy a single ZK, if it's possible, it wouldn't be something we recommend.

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+* _Tiered storage policies_ - Tiered storage offloads bookkeeper data to cheaper, long-term storage, and can impact cluster sizing. For more on tiered storage, see https://pulsar.apache.org/docs/2.11.x/tiered-storage-overview/[Pulsar documentation].
+
+There are other factors that could be a part of the aggregated cluster workload. As you gain familiarity with Pulsar you can further customize this calculation, but for now, we will estimate with the above numbers to size a cluster.
+
+[#assumptions]
+== Assumptions
+
+The methodology in this guide relies on the assumption that the ratio of the broker instance count to the bookkeeper instance count is static.
+For this example, we're using 1-to-5 as the static broker-to-bookkeeper ratio.
```
Comment (rodageve) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc]: With respect to static broker-to-bk ratio:

In Astra Streaming production clusters, we have (brokers to BKs) - 3-3, 5-3, 5-6.  Our Azure eastus cluster has the highest messaging workloads (small msg size), this is being used by our customer PEX.   Typically, we have more Brokers than BKs, but it does depend on workloads, and the resources given to the BKs (memory, cpu).  BKs performance is primarily disk-bound.

File: modules/install-upgrade/pages/production-cluster-vm.adoc [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+|5
+|5
+|2/2/1
+.5+a|* 3 AZs +
+* At least 1 Pulsar server instance per AZ +
+* Even distribution of Pulsar server instances across AZs
+
+|Bookkeeper
+|36
+|13/13/13
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-vm.adoc]: This should be 12

File: modules/install-upgrade/pages/production-cluster-vm.adoc [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+|*Value*
+
+|Availability Zones per region (AZs)footnote:[Pulsar server instances (of the same component type) should be evenly distributed across 3 AZs as much as possible, with minimum 1 Pulsar server instance per component type.]
+|3
+
+|Required Pulsar server components
+|Zookeepers, Bookkeepers, Brokers, Standalone autorecovery, Pulsar Proxy
+
+|Broker to bookkeeper ratio
+|1-to-5
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-vm.adoc]: Indicate that this is just a very simple assumption just to make the broker sizing calculation easy.

In order to get the accurate number of the brokers, a full fledged performance analysis is required in the provisioned environment with all hardware specification being considered, including the network link specification.

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+Pulsar clusters come in many shapes and sizes. There are minimum components for base functionality, and there are some recommended add-on components that make management and observability easier. For this guide we will focus on the required components and what it takes to make them resilient to outages and highly available in a three-zone cloud.
+
+=== Required components
+
+* Zookeeper - This is Pulsar’s meta data store. It stores data about a cluster’s configuration, helps the proxy direct messages to a broker, and holds Bookie configurations. To star, use 1 instance of Zookeeper in your cluster and let the number of lookups delegate scaling from there.
+
+* Broker - This is Pulsar's message router.
+Ideally, each broker should be fully utilized without becoming a performance bottleneck.
+The best way to optimize this is through performance testing based on your cluster's workload characteristics.
+Start with 1 broker per cluster and scale as needed.
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc]: Similar to Rodrigo's comment, if we intend to use this as production recommendation, I wouldn't suggest saying "start with 1 and scale as needed ..." for all Pulsar server components.

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+Fortunately, the Pulsar broker is stateless, so adding or removing brokers has minimal performance impact on the cluster.
+
+* Bookkeeper (bookie) - This is Pulsar’s data store.
+Bookkeeper stores message data in a low-latency, resilient way.
+Start with 3 bookies distributed across 3 availability zones.
+Pulsar uses Bookkeeper’s quorum math to function, so a loss of 1 Bookkeeper instance won’t bring your system down, but will cause some data loss.
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc:39]: This really depends on the settings of En/Qw/Qa. It doesn't necessarily to be the case that loosing one instance won't bring down the system. I would rather not mentioning this at all. This document is not intended to explain how Pulsar works.

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+* Auto recovery - This is a Pulsar component that recovers Bookkeeper data in the event of a Bookkeeper outage.
+* Pulsar Proxy - This is an optional proxy for routing client requests to brokers.
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc]: For on-prem VM based deployment, although "Pulsar Proxy" can be called as optional, but it is closer to not deployed at all. But for K8s based deployment, it is kind of mandatory.

So, I would suggest making clear of the distinction here.

File: modules/install-upgrade/pages/production-cluster-sizing.adoc [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+* Dedicated or combined functions worker(s) - Your cluster may or may not use functions workers, and may or may not run on the same machine as the broker.
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-sizing.adoc]: For "production" oriented deployment, we always want to use dedicated function workers, not the combined approach.

File: modules/install-upgrade/partials/minimum-sizing.adoc [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+|===
+|Environment
+|Server Component
+|Node Count
+|Note
+
+.2+|DEV
+|(combined) Zookeeper, Broker, and Bookie
+|3
+|1 node per AZ +
```
Comment (yabinmeng) [inline modules/install-upgrade/partials/minimum-sizing.adoc]: for DEV deployment, we doesn't necessarily follow this 1 node per AZ setup

File: modules/install-upgrade/partials/hardware-sizing.adoc [hunk 10]
```diff
@@ -0,0 +1,68 @@
+[cols=4*,options=header]
```
Comment (yabinmeng) [inline modules/install-upgrade/partials/hardware-sizing.adoc]: Some numbers are not the same as in the latest "Pulsar Cluster Deployment Reference Architecture and Recommended Production Settings" document. Please stop using the numbers in the previously shared customer specific document.
The numbers in the new doc are more aligned with our AS setup.

File: modules/install-upgrade/pages/production-cluster-multiregion.adoc [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+:clusterReplicationFactor: 2
+
+= Multi-region georeplication
+
+This page contains sizing and configuration information for a georeplicated VM-based production cluster. +
+Pulsar clusters can be configured to replicate data across multiple regions to maximize availability.
+
+We apply our sizing methodology with the following assumptions: +
+
+* Replication will be *asynchronous* and *will not* use a global Zookeeper.
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-multiregion.adoc]: What does it mean by "asyn replication"? Do we explain it somewhere? If so, can we provided a link?

File: modules/install-upgrade/pages/production-cluster-multiregion.adoc [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+| Number of cluster replications
+| {clusterReplicationFactor}
+
+|===
+
+With the aggregated workload characteristics, we apply our methodology to these characteristics to size a production cluster. +
+[source,plain,subs="attributes+"]
+----
+Total message size (raw) =
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-multiregion.adoc:58]: Is it possible to align the formula? The output is a bit messy

File: modules/install-upgrade/pages/production-cluster-multiregion.adoc [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+----
+
+.. But wait! We need to account for replicating this cluster across two regions.
+Following our assumptions above, we multiply the total message size by the number of regions we are replicating to, which in this case is {clusterReplicationFactor} regions.
+
+[source,plain,subs="attributes+"]
+----
+Adjusted message size (raw) =
+25 TB * // total message size (raw)
+{clusterReplicationFactor} // replication factor
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-multiregion.adoc]: comment "// replication factor", I wouldn't call this as "replication factor", this is imply the "# of cluster"

File: modules/install-upgrade/pages/production-cluster-multiregion.adoc [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+This page contains sizing and configuration information for a georeplicated VM-based production cluster. +
+Pulsar clusters can be configured to replicate data across multiple regions to maximize availability.
+
+We apply our sizing methodology with the following assumptions: +
+
+* Replication will be *asynchronous* and *will not* use a global Zookeeper.
+* We will replicate *all topics* across two regions.
+* We will assume the same time-to-live and retention policies as the initial Pulsar cluster (the cluster being replicated from).
+
+[#aggregate-worksheet]
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-multiregion.adoc]: Generally speaking, I really don't think this calculation is needed.
As long as we have the calculation of one cluster (in another doc), the sizing for a geo-replication case is simply times the number of clusters. Repeating the same calculation step is not necessary.

File: modules/install-upgrade/pages/production-cluster-vm.adoc [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+|===
+
+== Sizing calculation
+
+. Determine the Pulsar server instance counts for all required server component types.
+.. Multiply replication factor by average message payload size by average message throughput.
++
+[source,plain,subs="attributes+"]
+----
+Total message size (raw) =
```
Comment (yabinmeng) [inline modules/install-upgrade/pages/production-cluster-vm.adoc]: Can  we align the formula a bit? The output is a bit messy.

**Pull request merged**